### DISCUSSION TITLE
No distributed cache for an specific entry

### DISCUSSION THREAD

**Discussion by edcastanohunter** [ZiggyCreatures/FusionCache#90] [General]
Hi FusionCache Team, 

QQ: If I do want to use a second layer for many of the cache entries, but I do not want to go to the second layer in a specific key, what can I do?. I have set this 

`options.DistributedCacheDuration = TimeSpan.FromSeconds(0); `

but not sure if that goes to the DistributedCached.

Target version is: ZiggyCreatures.FusionCache V 0.16.0

**Comment by jodydonetti** [comment]
Hi @edcastanohunter , and thanks for using FusionCache!
I'm working on something related to this, that would also cover a couple of other needs pointed out by others in the community, but just to be sure what you mean: you would like some sort of flag to completely not consider the 2nd layer (distributed cache) for some calls, right? What I mean is both in the read phase **and** in the write phase, basically like pretending the 2nd layer is not there, but only for some calls, correct?

**Reply by jasenf** [reply]
We really like the way EasyCache works here.  At startup you initiate any number of different cache instances and types (local, redis, hybrid, fake) each with a label.

Then in your code you call something like GetCache('mylocalcache') and you are handed the proper cache you want to use.  

We have the exact scenario where we need items cached locally only, some in redis, and some local w redis as backplane only.  Works great.

**Reply by edcastanohunter** [reply]
Yes, that is the escenario, I think a flag would be the way to go. :)

**Reply by jodydonetti** [reply]
Hi @jasenf , long time no see 😬

> We really like the way EasyCache works here. At startup you initiate any number of different cache instances and types (local, redis, hybrid, fake) each with a label.
> Then in your code you call something like GetCache('mylocalcache') and you are handed the proper cache you want to use.

What you are describing is not what @edcastanohunter is talking about: here the scenario is having a cache configured with 2 layers (memory + distributed) and, on that same cache, "skip" the 2nd layer but only for some method calls.

What you are describing is multiple cache instances configured differently.

> We have the exact scenario where we need items cached locally only, some in redis, and some local w redis as backplane only. Works great.

I agree, such approach can work great, and you can do that too with FusionCache if you'd like.
Right now you can create how many different cache instances you want, all configured differently (with different levels, implementations, backplanes, etc) and all of them with a different name (see the `FusionCacheOptions.CacheName` option).

You can have something like:

```CSHARP
public static class MyCaches {
  public static FusionCache MyLocalCache = new FusionCache(new FusionCacheOptions() {
    CacheName = "MyLocalCache",
    DefaultEntryOptions = new FusionCacheEntryOptions {
      // ...
    }
  };

  public static FusionCache MyOtherCache = new FusionCache(new FusionCacheOptions() {
    CacheName = "MyOtherCache",
    DefaultEntryOptions = new FusionCacheEntryOptions {
      // ...
    }
  };

  // ...
}
```
Or, if you don't like singletons, just a `List<FusionCache>` + a method to do a lookup by `CacheName` or something like that.

The only thing that is currently missing is a way to integrate that with the DI part, but I'm experimenting with an approach similar to the one used with [named http clients](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-7.0#named-clients), so stay tuned 😉

**Reply by jodydonetti** [reply]
> Yes, that is the escenario, I think a flag would be the way to go. :)

Very good, so stay tuned for something about that very soon 😉

**Reply by jodydonetti** [reply]
Hi @edcastanohunter , I hope [this](https://github.com/ZiggyCreatures/FusionCache/issues/94) will cover your needs.
If you can, please give it a read and let me know what you think.
Thanks!

**Reply by edcastanohunter** [reply]
That will!, Thank you so much @jodydonetti

**Reply by jodydonetti** [reply]
Great! I will update you as soon as the next version will go live.

**Reply by jodydonetti** [reply]
Hi @edcastanohunter , [v0.18.0](https://github.com/ZiggyCreatures/FusionCache/releases/tag/v0.18.0) is out 🎉

It includes the new `SkipDistributedCache` option, please let me know if it works well for you.

Thanks!
